app/controllers/password_resets_controller.rb

```diff
@@ -0,0 +1,61 @@
+class PasswordResetsController < ApplicationController
+  before_action :get_user, only: [:edit, :update]
+  before_action :valid_user, only: [:edit, :update]
+  before_action :check_expiration, only: [:edit, :update]
+
+  def new; end
+
+  def edit; end
+
+  def create
+    @user = User.find_by(email: params[:password_reset][:email].downcase)
+    if @user
+      @user.create_reset_digest
+      @user.send_password_reset_email
+      flash[:info] = "Email sent with password reset instructions"
+      redirect_to root_url
+    else
+      flash.now[:danger] = "Email address not found"
+      render :new
+    end
+  end
+
+  def update
+    if params[:user][:password].empty?
+      @user.errors.add(:password, "can't be empty")
+      render :edit
+    elsif @user.update(user_params)
+      log_in @user
+      @user.update(reset_digest: nil)
+      flash[:success] = "Password has been reset."
+      redirect_to @user
+    else
+      render :edit
```

Comment:
trương hợp này lỗi khi nào e nhỉ? Nên bắn ra message nữa để người dùng biết.

Reply:
![image](https://github.com/DouTee09/SimpleApp/assets/117714011/29ecb476-a367-44e5-9810-39ab6bbc08e2)
if params[:user][:password].empty? đây anh

Reply:
chỗ 
```
else 
render :edit
flash[:error] = ... 
```
Expected của anh là thêm message vào đây? Lỗi như thế nào?

Reply:
@DouTee09

Reply:
else
      flash[:success] = "A failed update due to an invalid password!"
      render :edit
    end